pkg/state/chart_dependency.go

```diff
@@ -89,7 +93,7 @@ func (d *UnresolvedDependencies) contains(dep unresolvedChartDependency) bool {
 		return false
 	}
 	for _, existDep := range deps {
-		if existDep.ChartName == dep.ChartName {
+		if reflect.DeepEqual(existDep, dep) {
```

Comment:
Let me think a bit on implications on doing this, but theoretically, this should check only for "ID"s of two dependencies.

Previously, the chart name was ID. Now, we can selectively use either chart or alias as ID. So, this check should be written:

```suggestion
		if existingDep.ChartName == dep.ChartName && existingDep.Alias == dep.Alias {
```